protocol/communities/manager.go

```diff
@@ -1601,54 +1685,85 @@ func (m *Manager) AcceptRequestToJoin(request *requests.AcceptRequestToJoinCommu
 	if err != nil {
 		return nil, err
 	}
 
-	revealedAccounts, err := m.persistence.GetRequestToJoinRevealedAddresses(dbRequest.ID)
+	community, err := m.GetByID(dbRequest.CommunityID)
 	if err != nil {
 		return nil, err
 	}
 
-	permissionsSatisfied, isAdmin, err := m.accountsSatisfyPermissionsToJoin(community, revealedAccounts)
-	if err != nil {
-		return nil, err
-	}
+	if community.HasPermissionToSendCommunityEvents() {
+		if dbRequest.MarkedAsPendingByPrivilegedAccount() {
+			// if the request is in any pending state, it means our admin node has either
+			// already made a decision in the past, or previously received a decision by
+			// another admin, which in both cases means we're not allowed to override this
+			// state again
+			return nil, errors.New("request to join is already in pending state")
+		}
 
-	if !permissionsSatisfied {
-		return community, ErrNoPermissionToJoin
-	}
+		// admins do not perform permission checks, they merely mark the
+		// request as accepted (pending) and forward their decision to the control node
+		acceptedRequestsToJoin := make(map[string]*protobuf.CommunityRequestToJoin)
+		acceptedRequestsToJoin[dbRequest.PublicKey] = dbRequest.ToCommunityRequestToJoinProtobuf()
 
-	memberRoles := []protobuf.CommunityMember_Roles{}
-	if isAdmin {
-		memberRoles = []protobuf.CommunityMember_Roles{protobuf.CommunityMember_ROLE_ADMIN}
-	}
+		adminChanges := &CommunityEventChanges{
+			AcceptedRequestsToJoin: acceptedRequestsToJoin,
+		}
 
-	pk, err := common.HexToPubkey(dbRequest.PublicKey)
-	if err != nil {
-		return nil, err
-	}
+		err := community.addNewCommunityEvent(community.ToCommunityRequestToJoinAcceptCommunityEvent(adminChanges))
+		if err != nil {
+			return nil, err
+		}
 
-	_, err = community.AddMemberWithRevealedAccounts(dbRequest, memberRoles, revealedAccounts)
-	if err != nil {
-		return nil, err
+		if err := m.markRequestToJoinAsAcceptedPending(pk, community); err != nil {
+			return nil, err
+		}
 	}
 
-	channels, err := m.accountsSatisfyPermissionsToJoinChannels(community, revealedAccounts)
-	if err != nil {
-		return nil, err
-	}
+	if community.IsControlNode() {
+		revealedAccounts, err := m.persistence.GetRequestToJoinRevealedAddresses(dbRequest.ID)
+		if err != nil {
+			return nil, err
+		}
 
-	for channelID := range channels {
-		_, err = community.AddMemberToChat(channelID, pk, memberRoles)
+		permissionsSatisfied, isAdmin, err := m.accountsSatisfyPermissionsToJoin(community, revealedAccounts)
 		if err != nil {
 			return nil, err
 		}
-	}
 
-	if err := m.markRequestToJoin(pk, community); err != nil {
-		return nil, err
+		if !permissionsSatisfied {
+			return community, ErrNoPermissionToJoin
+		}
+
+		memberRoles := []protobuf.CommunityMember_Roles{}
+		if isAdmin {
+			memberRoles = []protobuf.CommunityMember_Roles{protobuf.CommunityMember_ROLE_ADMIN}
+		}
+
+		// TODO: Don't add member to community, instead store revealed accounts in database
```

Comment:
> They need to be store in a separated database table instead of the community descriptoin so they aren't shared with public metadata.

How will clients then construct the member list?

Reply:
Ah sorry, the comment has a mistake. This is not about the members but about the revealed addresses